### PR TITLE
Feature/search serialize service unit count

### PR DESCRIPTION
**Pull request by juuso-j**
# Unit count to services and servicenodes, also modularization of the api.

## Description
Now serializes the unit count per municipality for services and servicenodes. 

-----------------------------------------------------------------------------------------------
#### Search
 1. services/search/api.py
     * Added unit count for every municipality for ServieNodes and Services. Moved some functionality to utils.py
 2. services/search/constants.py
     * Created file and Added the contains constans used in search.     
 3. services/search/utils.py
     * Created utils.py and moved some of the search functionality to utils, added functions for servicenode and service that sets the unit_counts field in serialization. Added function that sets the fields for addresses in serialization
     
#### Model
 1. services/models/service_node.py
     * Function that return the Units for the ServiceNode

### CODE REVIEW

**Review by aceViilee — APPROVED** [review]
LGTM

File: services/search/api.py [hunk 1]
```diff
@@ -83,7 +84,45 @@ def to_representation(self, obj):
             return representation
 
         if object_type == "servicenode":
-            representation["ids"] = self.context["service_node_ids"][str(obj.id)]
+            ids = self.context["service_node_ids"][str(obj.id)]
+            representation["ids"] = ids
+            unit_counts = {}
+            print(ids)
```
Comment (mhieta) [inline services/search/api.py]: Please remove debug-prints

File: services/search/api.py [hunk 2]
```diff
@@ -83,7 +84,45 @@ def to_representation(self, obj):
...
+                    service_node_id=ids[0]
+                )
+                for service_node_count in service_node_count_qs:
+                    if hasattr(service_node_count.division, "name"):
+                        division = service_node_count.division.name_fi.lower()
+                    else:
+                        continue
+
+                    count = service_node_count.count
+                    print(division, " ", count)
```
Comment (mhieta) [inline services/search/api.py]: Please remove debug-prints

File: services/search/api.py [hunk 3]
```diff
@@ -83,7 +84,45 @@ def to_representation(self, obj):
...
+                unit_qs = Unit.objects.filter(id__in=units).distinct()
+                for unit in unit_qs:
+                    division = unit.municipality_id
+                    if not division:
+                        continue
+                    if division in unit_counts:
+                        unit_counts[division] += 1
+                    else:
+                        unit_counts[division] = 1
+                # breakpoint()
```
Comment (mhieta) [inline services/search/api.py]: Unnecessary comment

File: services/search/api.py [hunk 4]
```diff
@@ -83,7 +84,45 @@ def to_representation(self, obj):
...
+                    if division in unit_counts:
+                        unit_counts[division] += count
+                    else:
+                        unit_counts[division] = count
+            else:
+                units = []
+                for id in ids:
+                    service_node = ServiceNode.objects.get(id=id)
+                    units += service_node.get_units()
+                unit_qs = Unit.objects.filter(id__in=units).distinct()
```
Comment (mhieta) [inline services/search/api.py]: Not 100% sure, but could this be simplified to:
`unit_qs = units.distinct()` ?

**Review by mhieta — CHANGES_REQUESTED** [review]
Added couple of comments. Looks good otherwise.

**Review by mhieta — APPROVED** [review]
LGTM 👍

File: services/search/api.py (hunk 4 above)
Comment (mhieta) [reply services/search/api.py]: Good now.

**Pull request merged**